narwhals/_arrow/dataframe.py

```diff
@@ -700,23 +689,13 @@ def is_duplicated(self: Self) -> ArrowSeries:
             .group_by(self.columns)
             .aggregate([(col_token, "min"), (col_token, "max")])
         )
-        return ~ArrowSeries(
-            pc.and_(
-                pc.is_in(row_index, keep_idx[f"{col_token}_min"]),
-                pc.is_in(row_index, keep_idx[f"{col_token}_max"]),
-            ),
-            name="",
-            backend_version=self._backend_version,
-            version=self._version,
-        )
-
-    def is_unique(self: Self) -> ArrowSeries:
-        from narwhals._arrow.series import ArrowSeries
-
-        is_duplicated = self.is_duplicated()._native_series
-
         return ArrowSeries(
-            pc.invert(is_duplicated),
+            pa.chunked_array(
+                pc.and_(
+                    pc.is_in(row_index, keep_idx[f"{col_token}_min"]),
+                    pc.is_in(row_index, keep_idx[f"{col_token}_max"]),
+                )
+            ),
```

Comment:
pyarrow-stubs would have detected that this should have been `chunked_array` instead of `BooleanArray` 🙀 

one more reason (as if any were necessary) to use as many type checking features as are feasible

Reply:
yep same things I was seeing locally when running `mypy` outside of pre-commit :(